js/script.js

```diff
@@ -336,10 +359,37 @@ const showInputWrapper = () => {
     $inputWrapper.classList.toggle("hide");
 };
 
+const handleFilteredTasks = () => {
+    let filteredTasks = tasks;
+
+    if ($searchInput.value !== "") {
+        filteredTasks = searchTasks($searchInput.value);
+    }
+
+    if (filterBy === "incomplete") {
+        filteredTasks = filteredTasks.filter((task) => !task.isDone);
+    } else if (filterBy === "completed") {
+        filteredTasks = filteredTasks.filter((task) => task.isDone);
+    }
+
+    renderTasks(filteredTasks);
+};
+
+const handleSearchButton = () => {
+    $searchInput.value = "";
+    $searchInput.hidden = !$searchInput.hidden;
+    if (!$searchInput.hidden) {
+        $searchInput.focus();
+    } else {
+        renderTasks(tasks);
+    }
+};
```

Comment:
@ich-wtag bhai changed it

Reply:
Yes we need to render again because if input contains value and when we close the input field then we are setting the input to empty string so we need to render the list.